src/__experimental__/components/select/select.component.js

```diff
@@ -41,6 +53,12 @@ class Select extends React.Component {
 
   input = {} // this will store a reference to the input html element
 
+  componentWillReceiveProps(nextProps) {
```

Comment:
Is the `componentWillReceiveProps` necessary in here? Only case when this `nextProps.value !== this.props.value` condition is met is when component is used in controlled way. And according to the `getValue` method it seems that `this.state.value` is not used in that case. Also I think we should use `UNSAFE_componentWillReceiveProps` because some of old lifecycle methods are deprecated. Or better dont use deprecated lifecycle methods at all.

Reply:
Good point. `componentWillReceiveProps()` isn't necessary. As you said, `state.value` is only used in `getValue()`, and only when the component is uncontrolled. I've removed `componentWillReceiveProps()` in 68512b0 :thumbsup: